frontend/graphwidget.cpp

```diff
@@ -10,123 +10,220 @@
+
+void GraphWidget::getBinWidths()
+{
+    binWidths_.clear();
+    for (auto run : chartData_)
+    {
+        QVector<double> binWidths;
+        auto runArray = run.toArray();
+        for (auto i = 0; i < runArray.count() - 1; i++)
+        {
+            double binWidth = runArray.at(i + 1)[0].toDouble() - runArray.at(i)[0].toDouble();
+            binWidths.append(binWidth);
+        }
+        binWidths_.append(binWidths);
+    }
+}
+
 ChartView *GraphWidget::getChartView() { return ui_->chartView; }
 
-void GraphWidget::on_binWidths_clicked(bool checked)
+void GraphWidget::on_binWidths_toggled(bool checked)
 {
-    checked ? ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts/ &#181;s")
-            : ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts");
-    for (auto *series : ui_->chartView->chart()->series())
+    if (checked)
+    {
+        toggleOptions("binWidths");
+        ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts/ &#181;s");
+    }
+    else
+        ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts");
+    for (auto i = 0; i < ui_->chartView->chart()->series().count(); i++)
     {
-        auto xySeries = qobject_cast<QXYSeries *>(series);
+        auto xySeries = qobject_cast<QXYSeries *>(ui_->chartView->chart()->series()[i]);
         auto points = xySeries->points();
         xySeries->clear();
         if (checked)
-        {
-
-            for (auto i = 0; i < points.count() - 1; i++)
-            {
-                auto binWidth = points[i + 1].x() - points[i].x();
-                points[i].setY(points[i].y() / binWidth);
-            }
-        }
+            for (auto j = 0; j < points.count(); j++)
+                points[j].setY(points[j].y() / binWidths_[i][j]);
         else
-        {
-            for (auto i = 0; i < points.count() - 1; i++)
-            {
-                auto binWidth = points[i + 1].x() - points[i].x();
-                points[i].setY(points[i].y() * binWidth);
-            }
-        }
+            for (auto j = 0; j < points.count(); j++)
+                points[j].setY(points[j].y() * binWidths_[i][j]);
         xySeries->append(points);
     }
 }
 
-void GraphWidget::on_muAmps_clicked(bool checked)
+void GraphWidget::toggleOptions(QString option)
 {
-    checked ? ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts/ Total &#181;Amps")
-            : ui_->chartView->chart()->axes(Qt::Vertical)[0]->setTitleText("Counts");
-    emit test(checked);
+    if (ui_->binWidths->isChecked() && option != "binWidths")
+        ui_->binWidths->toggle();
+    if (ui_->muAmps->isChecked() && option != "muAmps")
+        ui_->muAmps->toggle();
+    if (ui_->runDivide->isChecked() && option != "runDivide")
+        ui_->runDivide->toggle();
+    if (ui_->monDivide->isChecked() && option != "monDivide")
+        ui_->monDivide->toggle();
```

Comment:
This aspect would be better implemented using `QRadioButton`s. In general the layout needs tidying - I'll send you a UI file with my suggestion in it.